### PR TITLE
LPS-199218 Create async & headles item actions

### PR DESCRIPTION
**Pull request by juanjofgliferay**
## Details
Related story/task: [LPS-191485](https://liferay.atlassian.net/browse/LPS-191485) / [LPS-199218](https://liferay.atlassian.net/browse/LPS-199218)
Figma: [Item actions designs](https://www.figma.com/file/87FEbztFRRf0OqJJhiqT0U/LPS-186881-configure-actions-for-dataset-views?type=design&node-id=224-44674&mode=design&t=shDvRzbSZ6gWBWHe-0)

### Changes from original design
- Headless type doesn't have the method picker/selector [see slack thread](https://liferay.slack.com/archives/C3JBR21HA/p1698057700420559?thread_ts=1697207711.213759&cid=C3JBR21HA)

## Goal
Admin users could define item actions of type `async` and `headless`

Needs `Feature Flag: LPS-194395`

## UI
[async-headless-actions.webm](https://github.com/liferay-frontend/liferay-portal/assets/132653342/5a90ce32-0190-41af-8981-0c223c44f984)

### CODE REVIEW

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/css/Actions.scss [hunk 1]
```diff
@@ -14,4 +14,8 @@
 			border-radius: 0;
 		}
 	}
+
```
Comment (juanjofgliferay) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/css/Actions.scss:17]: Changed due to https://github.com/liferay-frontend/liferay-portal/pull/3765#discussion_r1369851785

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/actions/ActionForm.tsx [hunk 2]
```diff
@@ -21,6 +22,23 @@ import openDefaultFailureToast from '../../utils/openDefaultFailureToast';
 import openDefaultSuccessToast from '../../utils/openDefaultSuccessToast';
 import {IFDSAction} from '../Actions';
 
+const ACTION_METHOD = {
+	DELETE: 'DELETE',
+	GET: 'GET',
+	PATCH: 'PATCH',
+	POST: 'POST',
+};
+
```
Comment (juanjofgliferay) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/actions/ActionForm.tsx:31]: Changed due to https://github.com/liferay-frontend/liferay-portal/pull/3765#discussion_r1369886284

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/actions/ActionForm.tsx [hunk 3]
```diff
@@ -44,6 +62,17 @@ const ACTION_TYPES = [
 	},
 ];
```
Comment (juanjofgliferay) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/actions/ActionForm.tsx:64]: Changed due to https://github.com/liferay-frontend/liferay-portal/pull/3765#discussion_r1369889907

**Review by markocikos — APPROVED** [review]
@juanjofgliferay LGTM! :sparkles:

**Comment by juanjofgliferay** [comment]
ci:test:relevant

**Comment by juanjofgliferay** [comment]
ci:test:relevant

**Comment by juanjofgliferay** [comment]
ci:test:relevant

**Comment by juanjofgliferay** [comment]
ci:test:stable

**Comment by juanjofgliferay** [comment]
ci:test:relevant

**Comment by juanjofgliferay** [comment]
ci:stop

**Comment by juanjofgliferay** [comment]
ci:test:relevant

**Comment by mateusralv** [comment]
Hey @juanjofgliferay and @markocikos the failures are not related. 
[
Testray Comparison ](https://testray.liferay.com/home/-/testray/runs/compare?formDate=1698414917381&testrayRunIdA=2447500155&testrayRunIdB=2447879259)

**Comment by markocikos** [comment]
ci:test:stable

**Comment by juanjofgliferay** [comment]
ci:test:stable

**Comment by juanjofgliferay** [comment]
ci:test:stable

**Comment by juanjofgliferay** [comment]
ci:test:stable

**Comment by juanjofgliferay** [comment]
ci:forward:force

**Comment by juanjofgliferay** [comment]
ci:forward:force

**Comment by juanjofgliferay** [comment]
ci:forward:force

**Pull request closed without merging**